### PR TITLE
Add compiler flags.

### PR DESCRIPTION
**Pull request by sellout**
* Turn on the long-disabled `-Ywarn-numeric-widen` after eliminating the
  warnings and
* add options to take advantage of Java 8 features (e.g., invoke dynamic
  & lambdas).

### CODE REVIEW

File: core/src/main/scala/quasar/fs/ManageFile.scala [hunk 1]
```diff
@@ -100,18 +91,12 @@ object ManageFile {
...
   object MoveScenario {
-    object Case {
-      final case class DirToDir(src: ADir, dst: ADir)
+    // FIXME: These constructors should be private, but that causes errors in
+    //        transformPaths.scala.
+    final case class DirToDir(src: ADir, dst: ADir)
         extends MoveScenario
-      final case class FileToFile(src: AFile, dst: AFile)
+    final case class FileToFile(src: AFile, dst: AFile)
         extends MoveScenario
```
Comment (sellout) [inline core/src/main/scala/quasar/fs/ManageFile.scala]: @wemrysi When I make these constructors private (and convert the calls in transformPaths.scala to use the prisms), I get errors like
```
monocle.Prism[quasar.fs.ManageFile.MoveScenario,(quasar.fs.ADir, quasar.fs.ADir)] does not take parameters
              (src, dst) => dirToDir(inPath(src), inPath(dst)),
                                    ^
```
yet, I managed to convert other types in this commit to that same style without problem. Any idea what I’m doing wrong?

Comment (wemrysi) [inline core/src/main/scala/quasar/fs/ManageFile.scala]: You'll need to import `quasar.fp.prism._` (or `quasar.fp._`) to get the implicits that provide the syntax.

Comment (sellout) [inline core/src/main/scala/quasar/fs/ManageFile.scala]: Ah, thanks. I was hoping it was something like that. I’ll add that to the Wiki page. Also, I don’t think we need `private[pkg]`, right, just `private` is fine. The only thing that needs to see the constructors are the prisms in the same object.

Comment (wemrysi) [inline core/src/main/scala/quasar/fs/ManageFile.scala]: Unfortunately, you do need `private[pkg]` as the prisms are in the _trait_'s object, not the subtype introduced by the constructor =(

Comment (wemrysi) [inline core/src/main/scala/quasar/fs/ManageFile.scala]: Though, and I didn't think about this before, it looks like `private[MoveScenario]` should work, which is a bit better.

**Comment by sellout** [comment]
Ok, this is ready for review, finally. Especially check the last changes to the build script.

**Comment by wemrysi** [comment]
:+1:

**Pull request merged**